stl/src/vector_algorithms.cpp

```diff
@@ -5,14 +5,14 @@
 #error _M_CEE_PURE should not be defined when compiling vector_algorithms.cpp.
 #endif
 
-#if defined(_M_IX86) || defined(_M_X64) // NB: includes _M_ARM64EC
+#if defined(_M_IX86) || defined(_M_X64) || defined(_M_ARM64) // NB: _M_X64 includes _M_ARM64EC

```

Comment:
IIUC this condition is always true and can be removed: at least one of `_M_IX86`, `_M_X64`, and `_M_ARM64` must be defined when compiling this file, because this STL only supports targeting x64 (which has `_M_X64` defined), x86 (`_M_IX86`), ARM64 (`_M_ARM64`), and ARM64EC (`_M_X64`).

Reply:
Aha good point, I have removed this now